script.embuary.helper/resources/lib/helper.py

```diff
@@ -227,29 +252,224 @@ def json_call(method,properties=None,sort=None,query_filter=None,limit=None,para
     if params is not None:
         json_string['params'].update(params)
 
-    json_string = json.dumps(json_string)
-    result = xbmc.executeJSONRPC(json_string)
-
-    if debug:
-        log('--> JSON CALL: ' + json_string)
-        log('--> JSON RESULT: ' + result)
+    jsonrpc_call = json.dumps(json_string)
+    result = xbmc.executeJSONRPC(jsonrpc_call)
 
     if not PYTHON3:
         result = unicode(result, 'utf-8', errors='ignore')
+    result = json.loads(result)
+
+    if debug:
+        log('--> JSON CALL: ' + json_prettyprint(json_string))
+        log('--> JSON RESULT: ' + json_prettyprint(result))
+
+    return result
+
 
-    return json.loads(result)
+def json_prettyprint(string):
+    return json.dumps(string, sort_keys=True, indent=4, separators=(',', ': '))
 
 
-def reload_widgets(instant=False,force=True):
-    log('Force widgets to refresh')
+def reload_widgets(instant=False,reason='Timer'):
+    log('Force widgets to refresh (%s)' % reason)
+
     timestamp = time.strftime('%Y%m%d%H%M%S', time.gmtime())
 
     if instant:
+        if condition('System.HasAlarm(WidgetRefresh)'):
+            execute('CancelAlarm(WidgetRefresh,silent)')
+
         winprop('EmbuaryWidgetUpdate', timestamp)
-        return
 
-    execute('AlarmClock(WidgetRefresh,SetProperty(EmbuaryWidgetUpdate,%s,home),00:05,silent)' % timestamp)
+    else:
+        execute('AlarmClock(WidgetRefresh,SetProperty(EmbuaryWidgetUpdate,%s,home),00:10,silent)' % timestamp)
+
+
+def sync_library_tags(tags=None,recreate=False):
+    save = False
+
+    if tags is None:
+        tags = get_library_tags()
+
+    try:
+        whitelist = eval(addon_data('tags_whitelist.' + xbmc.getSkinDir() +'.data'))
```

Comment:
Why do you read the content of a file and then call `eval()` on it? That seems like security risk? All commands will just be executed?

Reply:
because I store things like `[u'Dokus', u'Filme', u'Serien', u'Watchlist']` in the file and have to convert the string

Reply:
Why not just `import` those files?

Reply:
But doesn't that mean that I have to use a python file for storing? And importing it is the same "security risk" in my eyes.

Reply:
Ok, so why not just read it from a json file? No more risk then.

Reply:
Ok, can I put it on the list for the next version? Started a new job which is very time consuming. I don't think I can change it before the weekend.

Reply:
Changed to json